src/formatDate.js

```diff
@@ -50,7 +50,81 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const inputSeparator = fromFormat[fromFormat.length - 1];
+  const inputDateComponents = date.split(inputSeparator);
+
+  const outputSeparator = toFormat[toFormat.length - 1];
+  const outputDateComponents = [];
+
+  const dateComponents = {
+    day: '',
+    month: '',
+    year: '',
+    yearFormat: '',
+    dateFactory() {
+      for (let i = 0; i < fromFormat.length - 1; i++) {
+        switch (fromFormat[i]) {
+          case 'DD':
+            this.day = inputDateComponents[i];
+            break;
+          case 'MM':
+            this.month = inputDateComponents[i];
+            break;
+          case 'YY':
+          case 'YYYY':
+            this.year = inputDateComponents[i];
+            this.yearFormat = fromFormat[i];
+            break;
+          default:
+            console.error(`Wrong format: ${fromFormat[i]}`);
+        }
+      }
+    },
+    formatFactory() {
+      for (let i = 0; i < toFormat.length - 1; i++) {
+        const currentFormat = toFormat[i];
+
+        switch (currentFormat) {
+          case 'DD':
+            outputDateComponents.push(this.day);
+            break;
+          case 'MM':
+            outputDateComponents.push(this.month);
+            break;
+          case 'YY':
+          case 'YYYY':
+            const outputYear = this.convertYearFormat(
+              this.year,
+              this.yearFormat,
+              currentFormat
+            );
+
+            outputDateComponents.push(outputYear);
+            break;
+          default:
+            console.error(`Wrong format: ${currentFormat}`);
```

Comment:
The use of console.error in the dateComponents object's methods is not ideal. It is better to throw an error so that it can be handled properly in the calling function.